### PR TITLE
Standardize Framework Model Parameters

### LINKED ISSUE #1822: Use optional constructor arguments for framework models
#### Expected Behavior

Users should be able to simple say `new XXXModel()` and it works. This is the simplest case and the model should initialize with reasonable default parameter values. In addition, all parameters should be optional which will serve two purposes, first it allows the syntax above and second it allows for named parameters.

#### Actual Behavior

Existing models have their own various ways of ingesting parameters. Some have parameter classes some have constructors that require arguments, ect... The aim is standardize the pattern and make our approach w.r.t model parameters consistent.

#### Potential Solution

We've decided to use constructor arguments with all of them having reasonable default values.

#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

### PR DESCRIPTION
**Pull request by mchandschuh**
#### Description

Updates existing framework models to use optional constructor arguments.

#### Related Issue

Closes #1822 

#### Motivation and Context

We want instantiating these models to be easy and self documenting, By providing constructor argument default values we can achieve both.

#### Requires Documentation Change

No.

#### How Has This Been Tested?

Being a mechanical refactor of existing code, this was only briefly tested locally.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/ect)
Some constructor signatures were modified and a `Parameters` type was removed which could break user builds if they're referencing these types already (unlikely).

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm.Framework/Alphas/MacdAlphaModel.cs [hunk 1]
```diff
@@ -116,31 +127,17 @@ class SymbolData
...
-
-                Consolidator.DataConsolidated += OnDataConsolidated;
-            }
+                MACD = algorithm.MACD(security.Symbol, fastPeriod, slowPeriod, signalPeriod, movingAverageType);
 
-            /// <summary>
-            /// Cleans up the indicator and consolidator
-            /// </summary>
-            /// <param name="algorithm">The algorithm instance</param>
-            public void CleanUp(QCAlgorithmFramework algorithm)
```
Comment (jaredbroad) [inline Algorithm.Framework/Alphas/MacdAlphaModel.cs]: Removed on decision to reduce requirements of people using consolidators and self-cleaning up.

File: Algorithm.Framework/Alphas/MacdAlphaModel.cs [hunk 2]
```diff
@@ -30,22 +30,31 @@ namespace QuantConnect.Algorithm.Framework.Alphas
...
-        /// <param name="consolidatorPeriod">The period of the MACD's input</param>
-        /// <param name="insightPeriod">The period assigned to generate insight</param>
-        /// <param name="bounceThresholdPercent">The percent change required in the MACD signal line to warrant an up/down insight</param>
-        public MacdAlphaModel(TimeSpan consolidatorPeriod, TimeSpan insightPeriod, decimal bounceThresholdPercent)
+        /// <param name="fastPeriod">The MACD fast period</param>
+        /// <param name="slowPeriod">The MACD slow period</param>
+        /// <param name="signalPeriod">The smoothing period for the MACD signal</param>
+        /// <param name="movingAverageType">The type of moving average to use in the MACD</param>
+        public MacdAlphaModel(
+            int fastPeriod = 12,
```
Comment (jaredbroad) [inline Algorithm.Framework/Alphas/MacdAlphaModel.cs]: Updated constructor args to explicitly match the indicator arguments. Made them all inside 1 class file to make the alpha models easier to digest in a single take.

File: Algorithm.Framework/Alphas/RsiAlphaModel.cs [hunk 3]
```diff
@@ -157,27 +147,27 @@ public void OnSecuritiesChanged(QCAlgorithmFramework algorithm, SecurityChanges
         /// </summary>
         private State GetState(RelativeStrengthIndex rsi, State previous)
         {
-            if (rsi > _parameters.UpperRsiBound)
+            if (rsi > 70m)
```
Comment (jaredbroad) [inline Algorithm.Framework/Alphas/RsiAlphaModel.cs]: Harded coded for educational purposes.

File: Common/Time.cs [hunk 4]
```diff
@@ -221,6 +222,17 @@ public static DateTime Min(DateTime one, DateTime two)
             return one < two ? one : two;
         }
 
+        /// <summary>
+        /// Multiplies the specified interval by the multiplier
+        /// </summary>
+        /// <param name="interval">The interval to be multiplied, such as TimeSpan.FromSeconds(1)</param>
+        /// <param name="multiplier">The number of times to multiply the interval</param>
+        /// <returns>The multiplied interval, such as 1s*5 = 5s</returns>
+        public static TimeSpan Multiply(this TimeSpan interval, double multiplier)
```
Comment (jaredbroad) [inline Common/Time.cs]: Help convert indicator numerical periods into timespans.

File: Algorithm.Framework/Alphas/MacdAlphaModel.cs [hunk 5]
```diff
@@ -66,16 +75,17 @@ public IEnumerable<Insight> Update(QCAlgorithmFramework algorithm, Slice data)
...
-                else if (normalizedSignal < -_bounceThresholdPercent)
+                else if (normalizedSignal < -BounceThresholdPercent)
                 {
                     direction = InsightDirection.Down;
                 }
 
-                var insight = new Insight(sd.Security.Symbol, InsightType.Price, direction, _insightPeriod);
+                var insightPeriod = sd.DataResolution.Multiply(_fastPeriod);
+                var insight = new Insight(sd.Security.Symbol, InsightType.Price, direction, insightPeriod);
                 if (insight.Equals(sd.PreviousInsight))
```
Comment (jaredbroad) [inline Algorithm.Framework/Alphas/MacdAlphaModel.cs]: Note for future - have to make sure generated time isn't compared here. Or include generated time and change this comparison method.

Comment (mchandschuh) [reply Algorithm.Framework/Alphas/MacdAlphaModel.cs]: fyi, opened issue to address this: https://github.com/QuantConnect/Lean/issues/1829

File: Common/Time.cs (hunk 4 above)
Comment (mchandschuh) [reply Common/Time.cs]: and just plain awesome :) kind of surprised the .net framework doesn't support this operation

**Pull request merged**